test-cases/tests/performance/k04-run-performance-test-against-rhoam.md

```diff
@@ -0,0 +1,98 @@
+
+7. Run the performance test suite
+
+   The way to do it is described in [MGDAPI-238](https://issues.redhat.com/browse/MGDAPI-238) and in [Austin's doc](https://docs.google.com/document/d/1NJBUsieRkBLnN2PMAF5cpaH7uXq9mZCx1JQaT9Ruytk/edit?usp=sharing). Use [trepel branch](https://gitlab.cee.redhat.com/trepel/3scale-py-testsuite/-/tree/performance_tests). To validate the advertised load use [rhsso_tokens](https://gitlab.cee.redhat.com/trepel/3scale-py-testsuite/-/blob/performance_tests/testsuite/tests/performance/apicast/smoke/template_rhsso_tokens.hf.yaml) benchmark. It is set to have 10% of login flow requests. You will need to change `maxSessions` (~6000), `usersPerSec` (~25 to validate 20M load), `duration`, `maxDuration`, and [http.sharedConnections](https://gitlab.cee.redhat.com/trepel/3scale-py-testsuite/-/blob/performance_tests/testsuite/tests/performance/apicast/smoke/test_smoke_rhsso_tokens.py#L54-55) (~1200).
+
+- create a `perf-test-start-time.txt` file as described in [capture_resource_metrics script](https://github.com/integr8ly/integreatly-operator/blob/master/scripts/capture_resource_metrics.sh).
+  - the actual performance test run doesn't start immediately, the performance test suite creates various 3scale (Product, Backend, Application, Application Plan) and SSO (realm, client, users) entities first
+  - best to track the log of Hyperfoil controller to get the exact time the when `rampUp` phase starts
+
+8. Create a `perf-test-end-time.txt` for [capture_resource_metrics script](https://github.com/integr8ly/integreatly-operator/blob/master/scripts/capture_resource_metrics.sh)
+
+- to get the exact time track the Hyperfoil controller log
+
+9. Collect the data about the performance test run
+
+- from [alerts-during-perf-testing](https://github.com/integr8ly/integreatly-operator/blob/master/scripts/alerts-during-perf-testing.sh)
+- from Hyperfoil
+  - install Hyperfoil locally
+  - bin/cli.sh
+  - connect <hyperfoil-url-without-protocol> -p <port-8090-is-default>
+  - runs # to see all the runs
+  - status <your-run-name>
+  - stats <your-run-name>
+  - export -f json -d . <your-run-name> # to export the data
+  - use [report tool](https://github.com/Hyperfoil/report) to generate the HTML out of the exported data
+- eye review of various Grafana Dashboards, see [this guide](https://docs.google.com/document/d/1KznoB-we73lGUViJApVHyBoIgh3xpgyak6ODAEAHbwk/edit?usp=sharing) on how to do it
+- use [capture_resource_metrics script](https://github.com/integr8ly/integreatly-operator/blob/master/scripts/capture_resource_metrics.sh) to get the data
+- create a spreadsheet with all the collected data
+  - see [example spreadsheet](https://docs.google.com/spreadsheets/d/1v_bZIk8B_thZi93hGBNiOOnbSix0gmpwy3LV_s4WFPw/edit?usp=sharing)
+- add information about the benchmark used
```

Comment:
what kind of information should be added?

Reply:
This was added just in case we change our benchmark (e.g. different payload, different lenght, etc). There is a plan to change the benchmark to use more 3scale entities (Product, Backend, Application, Application Plan...currently only one of each is created and typical customer has much more of them). I thought such info might be useful since it also complicated direct comparison.

Reply:
Basicaly it is not to forgot to put important info to `Details / Note` row in the spreadsheet

Reply:
re-worded